### DISCUSSION TITLE
I18n solution for static export

### DISCUSSION THREAD

**Discussion by tomdohnal** [vercel/next.js#14316] [Help]
Hi :)

Recently, I've been struggling with setting up i18n with the Next.js static export. (Namely the `i18next` library.)

What I'd like to achieve?
* I want to generate *multiple versions* of the app, one "version" per one language.
* I want to handle all the translations at *build time*. (I'd like to output static HTML files translated to different languages and *don't* want to include the translation logic in the bundle which gets shipped to the user)

Has anyone had any success achieving something similar? The closest solution I found is https://github.com/claudetech/node-static-i18n, which is unfortunately only for plain HTML (no JSX).

**Comment by leerob** [comment]
Check out the new [i18n RFC](https://github.com/vercel/next.js/discussions/17078) and give us your feedback! 🚀
